osu.Game/Online/Multiplayer/PlaylistItem.cs

```diff
@@ -64,8 +64,8 @@ public PlaylistItem()
 
         public void MapObjects(BeatmapManager beatmaps, RulesetStore rulesets)
         {
-            Beatmap.Value = apiBeatmap.ToBeatmap(rulesets);
-            Ruleset.Value = rulesets.GetRuleset(RulesetID);
+            Beatmap.Value ??= apiBeatmap.ToBeatmap(rulesets);
+            Ruleset.Value ??= rulesets.GetRuleset(RulesetID);
```

Comment:
So `MapObjects()` won't ever overwrite a non-null beatmap anymore? The commit that introduced these (536df07) has a `(for testing)` annotation in the message - is this meant to be final? This affects the abstract `RoomManager`, and both multiplayer modes as a consequence.

Reply:
Yes this is meant to be final. It's here for tests where the beatmap is set directly on `PlaylistItem`s.